nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/src/main/java/org/apache/nifi/cluster/coordination/http/endpoints/ProcessorRunStatusDetailsEndpointMerger.java

```diff
@@ -0,0 +1,100 @@
+        }
+
+        final ProcessorsRunStatusDetailsEntity responseEntity = clientResponse.getClientResponse().readEntity(ProcessorsRunStatusDetailsEntity.class);
+
+        // Create mapping of Processor ID to its schedule Summary.
+        final Map<String, ProcessorRunStatusDetailsEntity> scheduleSummaries = responseEntity.getRunStatusDetails().stream()
+            .collect(Collectors.toMap(entity -> entity.getRunStatusDetails().getId(), entity -> entity));
+
+        for (final NodeResponse nodeResponse : successfulResponses) {
+            final ProcessorsRunStatusDetailsEntity nodeResponseEntity = nodeResponse == clientResponse ? responseEntity :
+                nodeResponse.getClientResponse().readEntity(ProcessorsRunStatusDetailsEntity.class);
+
+            for (final ProcessorRunStatusDetailsEntity processorEntity : nodeResponseEntity.getRunStatusDetails()) {
+                final String processorId = processorEntity.getRunStatusDetails().getId();
+
+                final ProcessorRunStatusDetailsEntity mergedEntity = scheduleSummaries.computeIfAbsent(processorId, id -> new ProcessorRunStatusDetailsEntity());
+                merge(mergedEntity, processorEntity);
+            }
+        }
+
+        final ProcessorsRunStatusDetailsEntity mergedEntity = new ProcessorsRunStatusDetailsEntity();
+        mergedEntity.setRunStatusDetails(new ArrayList<>(scheduleSummaries.values()));
+        return new NodeResponse(clientResponse, mergedEntity);
+    }
+
+    private void merge(final ProcessorRunStatusDetailsEntity target, final ProcessorRunStatusDetailsEntity additional) {
+        PermissionsDtoMerger.mergePermissions(target.getPermissions(), additional.getPermissions());
+
+        final ProcessorRunStatusDetailsDTO targetSummaryDto = target.getRunStatusDetails();
+        final ProcessorRunStatusDetailsDTO additionalSummaryDto = additional.getRunStatusDetails();
+
+        // If name is null, it's because of permissions, so we want to nullify it in the target.
+        if (additionalSummaryDto.getName() == null) {
+            targetSummaryDto.setName(null);
+        }
+
+        targetSummaryDto.setActiveThreadCount(targetSummaryDto.getActiveThreadCount() + additionalSummaryDto.getActiveThreadCount());
+
+        final String additionalRunStatus = additionalSummaryDto.getRunStatus();
+        if (RunStatus.Running.name().equals(additionalRunStatus)) {
+            targetSummaryDto.setRunStatus(RunStatus.Running.name());
+        } else if (RunStatus.Validating.name().equals(additionalRunStatus)) {
+            targetSummaryDto.setRunStatus(RunStatus.Validating.name());
+        } else if (RunStatus.Invalid.name().equals(additionalRunStatus)) {
+            targetSummaryDto.setRunStatus(RunStatus.Invalid.name());
+        }
+
+        // If validation errors is null, it's due to eprmissions, so we want to nullify it in the target.
+        if (additionalSummaryDto.getValidationErrors() == null) {
```

Comment:
Typo: eprmissions

Again, could use `!additional.getPermissions().getCanRead()` instead.
We may be relying on a "secondary" truth when we have access to the "primary".

Also to me it looks like the check could be incorrect in this way. Wouldn't the `validationErrors` be set to `null` if the processor is in `VALIDATING` according to this logic:
```java
        dto.setValidationErrors(convertValidationErrors(processor.getValidationErrors()));
...
    protected Collection<ValidationResult> getValidationErrors(final Set<ControllerServiceNode> servicesToIgnore) {
        final ValidationState validationState = this.validationState.get();
        if (validationState.getStatus() == ValidationStatus.VALIDATING) {
            return null;
        }
...
    private Set<String> convertValidationErrors(final Collection<ValidationResult> validationErrors) {
        if (validationErrors == null) {
            return null;
        }
```

Reply:
Yes - if the Processor is Validating, its validation errors should be null.